docs/content/reference/server_configuration.md

```diff
@@ -8,13 +8,19 @@ weight: 1
 
 ## Configuring the Monkey Island
 
-The Monkey Island Server is configured by creating a `server_config.json` file.
+The Monkey Island Server is configured through the `server_config.json` file.
+
+{{% notice info %}}
+Refer to the [setup guides](../../setup/) to learn how to use
+the `server_config.json` file for each deployment.
+{{% /notice %}}
 
 ### Creating a configuration file
 
 Here's an example `server_config.json` with all options specified:
 ```json
 {
+  "node_port": 443,
```

Comment:
`ui_port` is too vague and this const doesn't mean "port on which the UI is served". It means "port on which javascript runtime is running". Renamed to "javascript_runtime_port", which is longer and more convoluted, but requires no refactoring in case we move to bun (there are no more alternatives).

Reply:
> ui_port is too vague and this const doesn't mean "port on which the UI is served"

Is the UI not served on this port? If I point my browser at `https://monkeyserver:443`, will I not get the UI?

> but requires no refactoring in case we move to bun (there are no more alternatives)

There are no more alternatives, _today_.

Reply:
> Is the UI not served on this port?

It is served on this port. But this value defines not the port where the UI is served, but the port that node.js uses. These are separate things. UI was served on port 5000 before, but we had no `"ui_port": 5000`. UI port does not exists there's flask port and node port. If we want to hide this fact in the configuration, we should name it `island_port`.

Reply:
>  But this value defines not the port where the UI is served, but the port that node.js uses. 

Isn't this the port that node.js uses to serve the UI?

> If we want to hide this fact in the configuration...

We shouldn't leak implementation details through the config to the user. 

> we should name it `island_port`

Is that the port Agents will use to contact the Island? `island_port` is ambiguous.

Reply:
> Isn't this the port that node.js uses to serve the UI?

Serving the UI is not the only thing node can/will do. This const is used to determine on which port node should run. Where UI is served is decided by us and non-configurable for now.

> Is that the port Agents will use to contact the Island? island_port is ambiguous.

Yes, it's the port agents will use, because 443 is less likely to be restricted than 5000 and node port is configurable, whereas flask isn't.

> We shouldn't leak implementation details through the config to the user.

This is not a detail, it's one of 3 services comprising the island: node, flask, mongodb
We either call it as it is - node port or we obfuscate the "node" part and call it "javascript runtime". Config is meant to configure the Island services, not provide documentation for the user to read.

Reply:
Let's talk on webex. I'm missing a few details and we'll solve this quicker.